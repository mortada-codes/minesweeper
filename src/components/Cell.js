import React,{memo,} from 'react';
import {  View,Text,TouchableOpacity,StyleSheet} from 'react-native';







export default  memo(({width,location,onPress,cell})=>(<TouchableOpacity style={[{width,height:width,},styles.cell]} disabled={cell?.visited}  onPress={()=>onPress(location)}>
    <View style={styles.cellContent}>
<Text style={styles.text} testID="CellValue" >{cell?.visited?cell.value:''}</Text>
    </View>
</TouchableOpacity>))


const styles = StyleSheet.create({

    cell:{
        backgroundColor:'gray',
        borderWidth:1,
        borderColor:'white',
        justifyContent:'center',
        alignItems:'center',
    },
    cellContent:{
        justifyContent:'center',
        alignItems:'center',
    },
    visitedCell:{

    },
    text:{
        color:'white',
        textAlignVertical:'center',  
    },

})