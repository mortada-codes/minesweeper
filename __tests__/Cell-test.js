/**
 * @format
 */

import 'react-native';
import React from 'react';
import Cell from '../src/components/Cell';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render, wait,  } from '@testing-library/react-native';


 
it('renders correctly', () => {
   const { container, asJSON } =  render(<Cell />);
   expect(asJSON()).toMatchSnapshot();

});

it('not visited cell text should be empty',  () => {
   const location = [0,0]
   const cell = {value:1,visited:false}
   const { container, getByTestId } =  render(<Cell location={location} cell={cell} />);
     
   expect( getByTestId("CellValue").props.children).toMatch('')

});


it(' visited cell text should equal 1 ',  () => {
   const location = [0,0]
   const cell = {value:1,visited:true}
   const { container, getByTestId } =  render(<Cell location={location} cell={cell} />);
   
   expect( getByTestId("CellValue").props.children).toBe(1)

});