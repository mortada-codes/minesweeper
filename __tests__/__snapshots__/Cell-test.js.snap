// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <
    style={
      Array [
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "gray",
          "borderColor": "white",
          "borderWidth": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textAlignVertical": "center",
          }
        }
        testID="CellValue"
      >
        
      </Text>
    </View>
  </>
</View>
`;
